eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1563,52 +1563,58 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
+              val shortIds = createShortIds(d.channelId, realScidStatus)
+              DATA_WAIT_FOR_DUAL_FUNDING_READY(commitments1, shortIds)
+            case d: DATA_WAIT_FOR_CHANNEL_READY => d.copy(commitments = commitments1)
+            case d: DATA_WAIT_FOR_DUAL_FUNDING_READY => d.copy(commitments = commitments1)
+            case d: DATA_NORMAL => d.copy(commitments = commitments1)
+            case d: DATA_SHUTDOWN => d.copy(commitments = commitments1)
+            case d: DATA_NEGOTIATING => d.copy(commitments = commitments1)
+            case d: DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT => d.copy(commitments = commitments1)
+            case d: DATA_CLOSING => d.copy(commitments = commitments1)
+          }
+          stay() using d1 storing()
+        case Left(_) => stay()
+      }
 
     case Event(w: WatchFundingConfirmedTriggered, d: PersistentChannelData) =>
-      log.info(s"funding txid=${w.tx.txid} has been confirmed")
-      val commitments1 = acceptFundingTxConfirmed(w, d)
-      val d1 = d match {
-        // NB: we discard remote's stashed channel_ready, they will send it back at reconnection
-        case d: DATA_WAIT_FOR_FUNDING_CONFIRMED =>
-          val realScidStatus = RealScidStatus.Temporary(RealShortChannelId(w.blockHeight, w.txIndex, commitments1.latest.commitInput.outPoint.index.toInt))
-          val shortIds = createShortIds(d.channelId, realScidStatus)
-          DATA_WAIT_FOR_CHANNEL_READY(commitments1, shortIds)
-        case d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED =>
-          val realScidStatus = RealScidStatus.Temporary(RealShortChannelId(w.blockHeight, w.txIndex, commitments1.latest.commitInput.outPoint.index.toInt))
-          val shortIds = createShortIds(d.channelId, realScidStatus)
-          DATA_WAIT_FOR_DUAL_FUNDING_READY(commitments1, shortIds)
-        case d: DATA_WAIT_FOR_CHANNEL_READY => d.copy(commitments = commitments1)
-        case d: DATA_WAIT_FOR_DUAL_FUNDING_READY => d.copy(commitments = commitments1)
-        case d: DATA_NORMAL => d.copy(commitments = commitments1)
-        case d: DATA_SHUTDOWN => d.copy(commitments = commitments1)
-        case d: DATA_NEGOTIATING => d.copy(commitments = commitments1)
-        case d: DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT => d.copy(commitments = commitments1)
-        case d: DATA_CLOSING => d // there is a dedicated handler in CLOSING state
-      }
-      stay() using d1 storing()
+      acceptFundingTxConfirmed(w, d) match {
+        case Right(commitments1) =>
+          log.info(s"funding txid=${w.tx.txid} has been confirmed")
+          val d1 = d match {
+            // NB: we discard remote's stashed channel_ready, they will send it back at reconnection
+            case d: DATA_WAIT_FOR_FUNDING_CONFIRMED =>
+              val realScidStatus = RealScidStatus.Temporary(RealShortChannelId(w.blockHeight, w.txIndex, d.commitments.latest.commitInput.outPoint.index.toInt))
```

Comment:
You are re-introducing the previous bug here, this really must use `commitments1`:
```suggestion
              val realScidStatus = RealScidStatus.Temporary(RealShortChannelId(w.blockHeight, w.txIndex, commitments1.latest.commitInput.outPoint.index.toInt))
```